packages/compiler/src/ml_parser/html_whitespaces.ts

```diff
@@ -90,8 +92,30 @@ export class WhitespaceVisitor implements html.Visitor {
       const tokens = text.tokens.map((token) =>
         token.type === TokenType.TEXT ? createWhitespaceProcessedTextToken(token) : token,
       );
+
+      // Fully trim message when significant whitespace is not preserved.
+      if (!this.preserveSignificantWhitespace && tokens.length) {
```

Comment:
nit (for more explicitness, this is a more common pattern in most recent compiler code):
```suggestion
      if (!this.preserveSignificantWhitespace && tokens.length > 0) {
```